lib/net/peer.js

```diff
@@ -2145,6 +2221,29 @@ class PeerOptions {
       this.banScore = options.banScore;
     }
 
+    if (options.nonces != null) {
+      assert(typeof options.nonces === 'object');
+      assert(typeof options.nonces.alloc === 'function');
+      assert(typeof options.nonces.has === 'function');
+      assert(typeof options.nonces.remove === 'function');
+      this.nonces = options.nonces;
+    }
+
+    if (options.peerTypes != null) {
+      assert(typeof options.peerTypes === 'object');
+      this.peerTypes = options.peerTypes;
+    }
+
+    if (options.peerTypesByVal != null) {
+      assert(typeof options.peerTypesByVal === 'object');
+      this.peerTypesByVal = options.peerTypesByVal;
+    }
+
+    if (options.type != null) {
+      assert(typeof options.type === 'number');
```

Comment:
We should assert this is an integer.

Reply:
Better yet, we should assert that this is one of our peer types.

Reply:
I think asserting that it is an integer would fragment the style of the codebase.

```bash
$ grep -rin "=== 'number'" lib | wc -l
97
```

```bash
$ grep -rin "Number.isInteger" lib | wc -l
0
```